### DISCUSSION TITLE
pwm_set_output_polarity

### DISCUSSION THREAD

**Discussion by tinkerbug** [khoih-prog/RP2040_PWM#21] [Q&A]
Is there a way to set the output of two PWM channels to have opposite polarity? The Pico hardware PWM library has a method pwm_set_output_polarity that allows you set one channel to always be off while the other is on and vice versa. I am looking to create an alternating current for wireless power transfer and need the two channels to be always opposite. I am hoping to use Arduino for compatibility with other libraries, though I tested the Pico option and it does work.

Thanks!

**Comment by khoih-prog** [comment]
HI @tinkerbug 

![RP2040_PWM_Pushpull](https://user-images.githubusercontent.com/57012152/215651796-0c750c1f-9470-4bc4-90a4-6cbeef036b78.jpg)

I've modified the library to enable push-pull mode as in your question. Please reply if this is what you'd like? Anything else you need to have ?

**Reply by tinkerbug** [reply]
That is awesome, thank you!

**Comment by khoih-prog** [comment]
Hi @tinkerbug 

The [**RP2040_PWM** v1.7.0](https://github.com/khoih-prog/RP2040_PWM/releases/tag/v1.7.0) has just been published to add the new `PushPull` mode. Your contribution is noted in [**Contributions and Thanks**](https://github.com/khoih-prog/RP2040_PWM#contributions-and-thanks)

Please don't hesitate to post issue or even discussion, to request enhancement, new features, etc

Best Regards,

---

### Releases v1.7.0

1. Add functions `setPWMPushPull_Int`, `setPWMPushPull` and `setPWMPushPull_Period` for the new `PushPull` mode. Check [**pwm_set_output_polarity** #21](https://github.com/khoih-prog/RP2040_PWM/discussions/21)
2. Add these examples to demo the new `PushPull` mode
  - [**PWM_PushPull**](https://github.com/khoih-prog/RP2040_PWM/tree/main/examples/PWM_PushPull)
  - [**PWM_PushPull_DynamicDC**](https://github.com/khoih-prog/RP2040_PWM/tree/main/examples/PWM_PushPull_DynamicDC)
  - [**PWM_PushPull_DynamicFreq**](https://github.com/khoih-prog/RP2040_PWM/tree/main/examples/PWM_PushPull_DynamicFreq)
3. Fix bug of half frequency when using `phaseCorrect` mode
4. Improve `README.md` so that links can be used in other sites, such as `PIO`

![PWM_PushPull](https://user-images.githubusercontent.com/57012152/215689225-09b083d1-1cde-4130-a8f0-6bd4a23e48c1.jpg)
